app/main.py

```diff
@@ -86,129 +89,27 @@
+    lancelot = knights_config.get("lancelot")
+    lancelot_prepare = Knight(lancelot)
+    lancelot_prepare.all_preparations()
 
-    # apply weapon
-    mordred["power"] += mordred["weapon"]["power"]
+    arthur = knights_config.get("arthur")
+    arthur_prepare = Knight(arthur)
+    arthur_prepare.all_preparations()
 
-    # apply potion if exist
-    if mordred["potion"] is not None:
-        if "power" in mordred["potion"]["effect"]:
-            mordred["power"] += mordred["potion"]["effect"]["power"]
+    mordred = knights_config.get("mordred")
+    mordred_prepare = Knight(mordred)
+    mordred_prepare.all_preparations()
 
-        if "protection" in mordred["potion"]["effect"]:
-            mordred["protection"] += mordred["potion"]["effect"]["protection"]
+    red_knight = knights_config.get("red_knight")
+    red_knight_prepare = Knight(red_knight)
+    red_knight_prepare.all_preparations()
 
-        if "hp" in mordred["potion"]["effect"]:
-            mordred["hp"] += mordred["potion"]["effect"]["hp"]
-
-    # red_knight
-    red_knight = knightsConfig["red_knight"]
-
-    # apply armour
-    red_knight["protection"] = 0
-    for a in red_knight["armour"]:
-        red_knight["protection"] += a["protection"]
-
-    # apply weapon
-    red_knight["power"] += red_knight["weapon"]["power"]
-
-    # apply potion if exist
-    if red_knight["potion"] is not None:
-        if "power" in red_knight["potion"]["effect"]:
-            red_knight["power"] += red_knight["potion"]["effect"]["power"]
-
-        if "protection" in red_knight["potion"]["effect"]:
-            red_knight["protection"] += red_knight["potion"]["effect"]["protection"]
-
-        if "hp" in red_knight["potion"]["effect"]:
-            red_knight["hp"] += red_knight["potion"]["effect"]["hp"]
-
-    # -------------------------------------------------------------------------------
-    # BATTLE:
-
-    # 1 Lancelot vs Mordred:
-    lancelot["hp"] -= mordred["power"] - lancelot["protection"]
-    mordred["hp"] -= lancelot["power"] - mordred["protection"]
-
-    # check if someone fell in battle
-    if lancelot["hp"] <= 0:
-        lancelot["hp"] = 0
-
-    if mordred["hp"] <= 0:
-        mordred["hp"] = 0
-
-    # 2 Arthur vs Red Knight:
-    arthur["hp"] -= red_knight["power"] - arthur["protection"]
-    red_knight["hp"] -= arthur["power"] - red_knight["protection"]
-
-    # check if someone fell in battle
-    if arthur["hp"] <= 0:
-        arthur["hp"] = 0
-
-    if red_knight["hp"] <= 0:
-        red_knight["hp"] = 0
-
-    # Return battle results:
-    return {
-        lancelot["name"]: lancelot["hp"],
-        arthur["name"]: arthur["hp"],
-        mordred["name"]: mordred["hp"],
-        red_knight["name"]: red_knight["hp"],
-    }
+    battle1 = Battle(lancelot_prepare, mordred_prepare)
+    battle2 = Battle(arthur_prepare, red_knight_prepare)
 
+    battle1.battle_round()
+    battle2.battle_round()
 
-print(battle(KNIGHTS))
+    return Battle.battle_result
```

Comment:
The function returns `Battle.battle_result`, which is a class variable. If you want to keep results of different battles separate, consider returning individual battle results instead of a shared class variable.